core/src/main/java/com/exadel/aem/toolkit/core/lists/utils/ListResourceUtils.java

```diff
@@ -0,0 +1,154 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.MapUtils;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceUtil;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+import org.apache.sling.jcr.resource.api.JcrResourceConstants;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.Page;
+import com.adobe.granite.ui.components.ds.ValueMapResource;
+
+import com.exadel.aem.toolkit.core.CoreConstants;
+import com.exadel.aem.toolkit.core.lists.ListConstants;
+import com.exadel.aem.toolkit.core.lists.models.SimpleListItem;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Contains methods for manipulation with List Resource
+ */
+class ListResourceUtils {
+
+    private static final Map<String, Object> LIST_PROPERTIES
+        = Collections.singletonMap(JcrResourceConstants.SLING_RESOURCE_TYPE_PROPERTY, "wcm/foundation/components/responsivegrid");
+
+    private static final List<String> PROPERTIES_TO_IGNORE = Arrays.asList(
+        "jcr:createdBy", "jcr:created", "cq:lastModified", "cq:lastModifiedBy", "jcr:lastModified", "jcr:lastModifiedBy",
+        "cq:lastReplicationAction", "cq:lastReplicatedBy", "cq:lastReplicated"
+    );
+
+    /**
+     * Creates a list resource under given parent
+     * @param resourceResolver Sling {@link ResourceResolver} instance used to create the list
+     * @param parent           List Page that holds list resource
+     * @return list Resource or {@code null} if {@link ResourceResolver} or {@code parent} is null
+     * @throws PersistenceException if list resource cannot be created
+     */
+    static Resource createListResource(ResourceResolver resourceResolver, Page parent) throws PersistenceException {
+        if (resourceResolver == null || parent == null) {
+            return null;
+        }
+
+        return resourceResolver.create(parent.getContentResource(), ListConstants.NN_LIST, LIST_PROPERTIES);
+    }
+
+    /**
+     * Create {@link com.exadel.aem.toolkit.core.lists.models.internal.ListItemModel} resource under {@code parent}
+     * container with given properties.
+     * @param resourceResolver Sling {@link ResourceResolver} instance used to create the list
+     * @param parent           Container for {@code listItem}'s.
+     * @param properties       {@code listItem} properties
+     * @throws PersistenceException if {@code listItem} cannot be created
+     */
+    static void createListItem(ResourceResolver resourceResolver, Resource parent, Map<String, Object> properties) throws PersistenceException {
+        Map<String, Object> withoutSystemProperties = getWithoutSystemProperties(properties);
```

Comment:
(cosmetic) A variable's name should not start with a preposition because this is not the way a common person _names_ objects. As an entity of language. a variable corresponds to an "item", an "object", a "thing". We name things like that: "I have a pencil, I see a cat, I feel happiness". We do not say "I feel withoutsorrow"
Not a great deal, just a note if you don't mind)